vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -689,21 +689,39 @@ def build_connector_meta(
                 kv_transfer_params=req.kv_transfer_params,
             )
 
-        for req_id, (req, block_ids) in self._reqs_need_save.items():
+        # NOTE: For prefill side, there might be chance that early added
+        # request is chunked prefill, so we need to check if new blocks are added
+        for req_id, new_block_id_groups, preempted in yield_req_data(scheduler_output):
+            if req_id not in self._reqs_need_save:
+                continue
+            if preempted:
+                self._reqs_need_save.pop(req_id)
```

Comment:
This will cause preempted requests to be excluded from sending.
I think this should be removed:
```suggestion
```

Reply:
Wondering if request was scheduled as preempted before finishing prefill, should we still let Decoder reading this?

Reply:
The decoder starts pulling data only after the prefiller finished with the request, right?
In that case, preemption on the profiller should not matter for the decoder.
It just means the request gets re-allocated inside the prefiller.

Reply:
Makes sense, I was still thinking about my old implementation that copy to host happens when entire request completes.  For the copy immediately strategy, you're right, if request gets preempted or not should not impact here,
I have removed above if statement, Thanks!